index.js

```diff
@@ -6,346 +6,298 @@ var https = require('https');
+      });
     };
 
-    config = defaults(config, defaultOptions);
+    app_directories(app_dir).forEach(function(dir) {
+      var package_json = path.join(app_dir, dir, "/package.json");
+      if (!fs.existsSync(package_json) || !fs.statSync(package_json).isFile()) {
+        self.error("   package.json not found in directory " + dir);
+        return;
+      }
+
+      var pkg = JSON.parse(fs.readFileSync(package_json, 'utf8'));
+      if (!pkg || !pkg.main || !pkg.name) {
+        self.error("   failed to load " + package_json);
+        return;
+      }
+
+      var main = fs.realpathSync(path.join(app_dir, dir, pkg.main));
+      if (!fs.existsSync(main) || !fs.statSync(main).isFile()) {
+        self.error("   main file not found for app [" + pkg.name + "]: " + main);
+        return;
+      }
+
+      var app;
+      try {
+        app = require(main);
+      } catch (e) {
+        self.error("   error loading app [" + main + "]: " + e);
+        return;
+      }
+
+      self.apps[pkg.name] = pkg;
+      self.apps[pkg.name].exports = app;
+      if (typeof app.express != "function") {
+        self.error("   app [" + pkg.name + "] is not an instance of alexa-app");
+        return;
+      }
+
+      // Extract Alexa-specific attributes from package.json, if they exist
+      if (typeof pkg.alexa == "object") {
+        app.id = pkg.alexa.applicationId;
+      }
+
+      // attach the alexa-app instance to the alexa router
+      app.express({
+        expressApp: alexaRouter,
+        router: express.Router(),
+        debug: config.debug,
+        checkCert: config.verify,
+        preRequest: config.preRequest,
+        postRequest: config.postRequest
+      });
+
+      self.log("   loaded app [" + pkg.name + "] at endpoint: " + normalizedRoot + "/" + pkg.name);
+    });
+
+    return self.apps;
+  };
+
+  // Load server modules. For example, code the process forms, etc. Anything that
+  // wants to hook into express
+  self.load_server_modules = function(server_dir) {
+    var server_files = function(srcpath) {
+      return fs.readdirSync(srcpath).filter(function(file) {
+        return fs.statSync(path.join(srcpath, file)).isFile();
+      });
+    };
+    server_files(server_dir).forEach(function(file) {
+      file = fs.realpathSync(path.join(server_dir, file));
+      self.log("   loaded " + file);
+      var func = require(file);
+      if (typeof func == "function") {
+        func(self.express, self);
+      }
+    });
+  };
+
+  self.start = function() {
+    // Instantiate up the server
+    // TODO: add i18n support (i18n-node might be a good look)
+    // Issue #12: https://github.com/alexa-js/alexa-app-server/issues/12
+    self.express = express();
+
+    // TODO: change this to make sure it doesn't affect other non-Alexa services/apps
+    // Issue #35: https://github.com/alexa-js/alexa-app-server/issues/35
+    self.express.use(bodyParser.urlencoded({ extended: true }));
```

Comment:
According to [this](https://github.com/alexa-js/alexa-app-server/pull/18#issuecomment-273465357) comment in #18, shouldn't it be this:
```javascript
self.express.use("/alexa", bodyParser.urlencoded({ extended: true }));
```
#35 stated that "there are many express routes added which don't use urlencoded forms". So logically, one would want to only bind with Alexa-based routes.